service/history/handler/handler.go

```diff
@@ -112,16 +112,19 @@ func NewHandler(
 
 // Start starts the handler
 func (h *handlerImpl) Start() {
-	h.replicationTaskFetchers = replication.NewTaskFetchers(
+	var err error
+	h.replicationTaskFetchers, err = replication.NewTaskFetchers(
 		h.GetLogger(),
 		h.config,
 		h.GetClusterMetadata(),
 		h.GetClientBean(),
 	)
+	if err != nil {
```

Comment:
consider moving `replicationTaskFetchers` initialization to constructor and return error from constructor